### PR TITLE
Nix flake

### PR DESCRIPTION
**Pull request by yajo**
Add precommix for linting and formatting.

- [x] Provide a dev shell
- [x] Provide a nix package
- [x] Provide a nix flake
- [x] Fix CI
- [x] Document installation
- [x] Document hacking
- [x] Provide as much architectures as possible

### CODE REVIEW

File: docs/configuring.md [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ The file must have this content:
 
 ```yaml
 # Changes here will be overwritten by Copier; NEVER EDIT MANUALLY
-{{_copier_answers|to_nice_yaml}}
+{ { _copier_answers|to_nice_yaml } }
```
[the remark this answers is not in the document]
Comment (yajo) [reply docs/configuring.md:1082]: Hmm indeed... 🤔 

I guess the new formatters are smarter than us. What's the correct language indicator here? `yaml+jinja`?

[the remark this answers is not in the document]
Comment (pawamoy) [reply docs/configuring.md:1082]: I think so, or the reverse, `jinja+yaml`.

File: .editorconfig [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
-insert_final_newline = true
+indent_size = 4
+end_of_line = lf
+charset = utf-8
 trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.py]
+# For isort
+profile = black
```
Comment (sisp) [inline .editorconfig:13]: According to the `isort` docs, [`.editorconfig` is supported](https://pycqa.github.io/isort/docs/configuration/config_files.html#editorconfig) but [`pyproject.toml` is preferred](https://pycqa.github.io/isort/docs/configuration/config_files.html#editorconfig).

I've replaced explicit `isort` settings by the profile for Black in `pyproject.toml` in #756.

Comment (yajo) [reply .editorconfig:13]: Yes, this PR doesn't follow the most pure approach on config files for now, but the most practical.

Precommix supports non-python projects, where a `pyproject.toml` is not necessary. So it uses other dotfiles instead if possible.

I hope the world of linters gravitates around https://dot-config.github.io/ soon and then these problems start disappearing.

**Comment by yajo** [comment]
FTR I added a poll for Copier contributors in https://github.com/copier-org/copier/discussions/757.

**Comment by yajo** [comment]
Cool! This is now working! To try it out:

```bash
nix shell github:copier-org/copier/precommix
```

That will load copier installed with nix in your shell. Now go play with it.

Pending: docs, fix CI (all should break right now), provide more arches.

**Comment by nialov** [comment]
> Cool! This is now working! To try it out:
> 
> ```shell
> nix shell github:copier-org/copier/precommix
> ```
> 
> That will load copier installed with nix in your shell. Now go play with it.
> 
> Pending: docs, fix CI (all should break right now), provide more arches.

Thank you for packaging `copier`! Using the exact flake you provided worked at first but I believe the inputs used some GitHub nomenclature related to pull requests so they stopped working (the inputs with # hack comments afterwards). I packaged it using exact `git` hashes here (https://github.com/nialov/nialov-py-template) and added `git` into the path of the `copier` executable as it seemed to error without it. Should work with `nix run github:nialov/nialov-py-template#copier`.

**Comment by rszamszur** [comment]
@yajo FYI there is a project that integrates pre-commit with Nix: https://github.com/cachix/pre-commit-hooks.nix

**Pull request merged**